_posts/2020-04-06-cicd-part-1.md

```diff
@@ -0,0 +1,180 @@
+
+The tooling that developers use both in and around the release process is on that I find most interesting. It is one of the most critical stages in the software process, yet is regularly the most brittle, insecure, and misunderstood tools in most organizations. For those of you at large companies, ask around your office: just how many people /really/ understand your build tooling? It's not uncommon for their to be only one to be one or two build 'wizards' that really understand the companies release process end-to-end. This is a shame for a few reasons. It is the one place that all work flows through; having knowledge and an appreciatiation of the process means that more developers can contribute to it's improvement, thereby rapidly accelerating the productivity of not only themselves, but their peers as well. Likewise, a good build and release process confers a good knowledge of the frameworks and tools you are using. This is knowledge that can be employed to be a more productive programmer in your day-to-day.
+
+I intend for this series to serve a few purposes;
+
+- To serve as introduction to some common types of build & release processes
+
+  It's entirely likely that you may spend an entire career and never release a library to public package repository. That said, having an understand of the process and challenges in doing so may help you in solving other problems. You may be struggling with to decide the best way to build and release your own projects, and I hope that you be able to derive some inspiration from the examples contained within.
+
+- To allow the read to observe a build & release process as it increases in sophistication
+
+  The education most receive in software delivery is one that comes via a trial-by-fire. You start with a small solution to a small problem and iterate from there. I think it was easy to do this in years gone by when times where simpler, but the fast-paced nature of the world has become fairly unforgiving, and there are fewer low-risk opportunities to really touch-on software release processes. Increasing levels of abstraction and the proliferation of managed services make it harder to gain context and to understand the 'why', which is neccesary when faced with obstacles that you may be unfamiliar with.
+
+- To understand that software is distributed and consumed in different ways, and this inevitably influences the software development process
+
+  Software is consumed in wildly different ways depending on the who it is for. Software is installed on desktops using binaries retrieved via FTP. Libraries are distributed via various package manager. SaaS is commonly delivered via web browser. How your software is delivered and consumed has a marked effect on your build & release process.
+
+- Evaluate different CI/CD processes within the lens of the AWS Well-Architected Framework
```

Comment:
Are we selling a WAF?

Reply:
Could we link it into the new ability to provide them? Do we have a page for that?